lms/djangoapps/support/static/support/jsx/single_support_form.jsx

```diff
@@ -183,8 +193,66 @@ class RenderForm extends React.Component {
     );
   }
 
+  async handleInputChange(event) {
+    event.preventDefault();
+    const queryString = event.target.value;
+    if (queryString.length > 2) {
+      let suggestions = [];
+      await fetch(`${this.learnerSupportCenterURL}/hc/api/internal/instant_search.json?query=${queryString}&locale=en-us`, { headers: {
+        'Content-Type': 'application/json',
+      }}).then((response) => {
+        suggestions = response.json.results;
+        this.setState({ suggestions });
+      });
+    } else {
+      this.setState({
+        suggestions: [],
+        activeSuggestion: 0,
+      });
+    }
+  }
+
+  onKeyDown(event) {
+    const { activeSuggestion, suggestions } = this.state;
+
+    if (event.keyCode === 13) { // User pressed the enter key
+      window.location.href = this.learnerSupportCenterURL + suggestions[activeSuggestion].url;
+    } else if (event.keyCode === 38) { // User pressed the up arrow
+      if (activeSuggestion === 0) { return; }
+
+      this.setState({ activeSuggestion: activeSuggestion - 1 });
+    } else if (event.keyCode === 40) { // User pressed the down arrow
+      if (activeSuggestion - 1 === suggestions.length) { return; }
+
+      this.setState({ activeSuggestion: activeSuggestion + 1 });
+    }
+  }
+
+  handleBlur(event) {
+    if (!this._ignoreBlur) {
+      this.setState({
+        suggestions: [],
+        activeSuggestion: 0,
+      });
+    }
+  }
+
+  handleSearchButton(event) {
+    event.preventDefault();
+    let queryString = document.getElementById('query').value;
+    queryString = queryString.replace(' ', '+');
+
+    window.location.href = `${this.learnerSupportCenterURL}/hc/en-us/search?utf8=✓&query=${queryString}`;
+  }
+
+  handleSuggestionClick(url) {
+    window.location.href = this.learnerSupportCenterURL + url;
+  }
```

Comment:
I would suggest a redirect instead of changing the current window. Also, make sure rel='noopener' is activated when doing the redirection.

Reply:
I used this to keep the consistency in this file as changing `window.location.href` is used before my changes.
I think I can use `this.props.history.push(url);` instead of Redirect component (which needs to be written in a render function)